applications/rag/variables.tf

```diff
@@ -0,0 +1,118 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+variable "project_id" {
+  type        = string
+  description = "GCP project id"
+}
+
+variable "cluster_name" {
+  type = string
+}
+
+variable "cluster_membership_id" {
+  type        = string
+  description = "require to use connectgateway for private clusters, default: cluster_name"
+  default     = ""
+}
+
+variable "cluster_location" {
+  type = string
+}
+
+variable "kubernetes_namespace" {
+  type        = string
+  description = "Kubernetes namespace where resources are deployed"
+  default     = "rag"
+}
+
+variable "jupyter_service_account" {
+  type        = string
+  description = "Google Cloud IAM service account for authenticating with GCP services"
+  default     = "jupyter-system-account"
+}
+
+variable "ray_service_account" {
+  type        = string
+  description = "Google Cloud IAM service account for authenticating with GCP services"
+  default     = "ray-system-account"
+}
+
+variable "rag_service_account" {
+  type        = string
+  description = "Google Cloud IAM service account for authenticating with GCP services"
+  default     = "rag-system-account"
+}
+
+variable "gcs_bucket" {
+  type        = string
+  description = "GCS bucket name to store dataset"
+}
+
+variable "default_backend_service" {
+  type    = string
+  default = "proxy-public"
+}
+
+variable "members_allowlist" {
+  type    = string
+  default = ""
+}
+variable "add_auth" {
+  type        = bool
+  description = "Enable iap authentication on jupyterhub"
+  default     = true
+}
+
+variable "service_name" {
+  type        = string
+  description = "Name of the Backend Service on GCP"
```

Comment:
Discussed with @umeshkumhar This description is wrong, actual we are using it as the name of kubernetes backend_config name